services/provider/client/client.go

```diff
@@ -24,23 +24,39 @@ func NewConsumerClient(cc *grpc.ClientConn, timeout time.Duration) *ConsumerClie
 func NewGRPCConnection(serverAddr string, opts []grpc.DialOption) (*grpc.ClientConn, error) {
 	conn, err := grpc.Dial(serverAddr, opts...)
 	if err != nil {
-		klog.Fatalf("failed to dial: %v", err)
+		return nil, fmt.Errorf("failed to dial: %v", err)
 	}
 	return conn, err
 }
 
-// OnBoardConsumer to validate the consumer and create StorageConsumer
+// Initialize initialzes a new provider client
+func Initialize(serverAddr, token string, reqTimeOut time.Duration) (*ConsumerClient, error) {
+	interceptor := NewAuthInterceptor(token)
+
+	// TODO: ADD TLS support
+	opts := []grpc.DialOption{
+		grpc.WithUnaryInterceptor(interceptor.Unary()),
+	}
+	cc, err := NewGRPCConnection(serverAddr, opts)
```

Comment:
can we also implement the logic to close the connection

Reply:
I realised that it won't be easy to add close() using  current approach.  So moved back to original changes. You use and close the connection like this : 
```
conn, err := client.NewGRPCConnection("address", "token")

cc := NewConsumerClient(conn, <timeout>)
// use cc

// close cc
cc.Close()
```